File: <100/008/008.py
num_digits = 13


def solve(s: str) -> int:
    N: int = len(s)
    m: int = 0
    for j in range(N-num_digits):
        tmp: int = 1
        for d in range(num_digits):
            tmp *= int(s[j+d])
        m = max(m, tmp)

    return m


def main() -> None:
    filename = "008.txt"
    s: str = ""
    with open(filename, 'r') as f:
        for row in f:
            if row[-1] == '\n':
                row = row[:-1]
            s += row

    print(solve(s))

    return


if __name__ == '__main__':
    main()

File: <100/008/008.jl
function readfile(filename:: String):: Array{String}
  f = open(filename, "r")
  arr:: Array{String} = readlines(f)
  close(f)
  
  return arr
end

function cat_strings(arr:: Array{String}):: String
  S:: String = ""
  for s ∈ arr
    S *= s
  end

  return S
end

function solve(S:: String):: Int
  answer:: Int = 0
  L:: Int = S |> length
  width:: Int = 13
  for j ∈ 1:(L-width)
    tmp:: Int = 1
    for c ∈ S[j : j + width-1]
      tmp *= parse(Int, c)
    end
    answer = max(answer, tmp)
  end

  return answer
end

function main():: Nothing
  filename = "008.txt"
  filename |> readfile |> cat_strings |> solve |> println
  return nothing
end


@time main()